### PR TITLE
Dark mode logo

### PR DESCRIPTION
**Pull request by segsell**
Add dark mode logo to the navigation bar and on the main index page.

- [x] switch to ```furo``` theme
- [x] navigation bar
- [x] index page
- [x] Remove white boxes around logos on grid card items

### CODE REVIEW

**Comment by segsell** [comment]
The main page logo in dark mode is causing trouble. I've tried a couple things (like making boxes dark/transparent) in ```index.md``` and/or ```custom.css```, but haven't been successful yet.

**Comment by segsell** [comment]
The link to the logo in the toolbar is not being found yet. Using the ```https://``` link to the pandas logo worked. Maybe, once we merge into main, the location of our dark mode logo will be found.

File: docs/rtd_environment.yml [hunk 1]
```diff
@@ -13,10 +13,12 @@ dependencies:
   - sphinxcontrib-bibtex
   - sphinx-copybutton
   - sphinx-panels
+  - sphinx-design
   - ipython
   - ipython_genutils
   - myst-nb
   - pydata-sphinx-theme
```
Comment (janosg) [inline docs/rtd_environment.yml:20]: Two things:

1. Can you do the same changes in the central [`environment.yml`](https://github.com/OpenSourceEconomics/estimagic/blob/main/environment.yml)
2. Can't we remove `pydata_sphinx_theme` from the environment?

**Review by janosg — COMMENTED** [review]
- I think the current dashboard is not pretty enough to be included on the front page of the documentation and the readme. Can you remove the gif in both places? We'll work on something better soon. 
- The toctree on [this page](https://estimagic--479.org.readthedocs.build/en/479/getting_started/estimation/index.html) is too deep. I think we only want to show one level, i.e. "likelihood estimation" and "method of simulated moments (MSM) estimation"
- When dark-mode is active, the plots in notebooks are still shown in a light theme. QuantEcon has found a way to switch to a dark theme. Click on the change contrast symbol [here](https://python-programming.quantecon.org/matplotlib.html) to try it out. Do you think it would be super hard to add this?

**Comment by janosg** [comment]
> The link to the logo in the navigation bar is not being found. Using the [pandas dark mode logo](https://github.com/pandas-dev/pandas/blob/main/doc/source/conf.py#L250) and our [standard estimagic logo](https://github.com/OpenSourceEconomics/estimagic/blob/main/docs/source/_static/images/estimagic_logo.svg) as dummies worked. Maybe, once we merge this PR into main, the location of our dark mode logo will be available, too. Note that the logo settings for the navigation bar are specified in `conf.py`. Maybe this makes it different to the main index page, which is set up in `index.md`. There, the same image reference is actually found.

Is this still the case? I did not see any problems.

**Comment by segsell** [comment]
> > The link to the logo in the navigation bar is not being found. Using the [pandas dark mode logo](https://github.com/pandas-dev/pandas/blob/main/doc/source/conf.py#L250) and our [standard estimagic logo](https://github.com/OpenSourceEconomics/estimagic/blob/main/docs/source/_static/images/estimagic_logo.svg) as dummies worked. Maybe, once we merge this PR into main, the location of our dark mode logo will be available, too. Note that the logo settings for the navigation bar are specified in `conf.py`. Maybe this makes it different to the main index page, which is set up in `index.md`. There, the same image reference is actually found.
> 
> Is this still the case? I did not see any problems.

That's outdated and has been fixed. Thanks!

**Comment by segsell** [comment]
>     * When dark-mode is active, the plots in notebooks are still shown in a light theme. QuantEcon has found a way to switch to a dark theme. Click on the change contrast symbol [here](https://python-programming.quantecon.org/matplotlib.html) to try it out. Do you think it would be super hard to add this?

Unfortunately, that's not straightforward. It seems like QuantEcon have developed their own docs template. I haven't seen any sphinx theme like ```furo``` or ```pydata``` that have this feature. A second difference is that they use a ```.md``` file instead of a ```.ipynb``` jupyter notebook. 

Changing the ```matplotlib``` theme in a jupyter notebook is straightforward (in a static way). However, automatically switching between the light and dark theme is tricky. Locally, the workaround is [modifying some ```.ipython``` file](https://medium.com/@rbmsingh/making-jupyter-dark-mode-great-5adaedd814db). At the moment, I don't have an idea how this can be achieved for the sphinx documentation.

**Review by janosg — APPROVED** [review]
Thanks!

**Pull request merged**